Update loader to work with whitelisted dss (resolves STOR-2)

jessebrennan opened the pull request:
This *should* have just worked out of the box without this PR but there were some bugs to resolve.

### bug 1.
The integration tests used the requests library to get some stuff. Once things become whitelisted and authenticated this becomes a problem. The fix was easy, just use the dss stuff so it all happens correctly behind the scene.

### bug 2.
The way the dss config was initialized caused some problems. The config remembers your setup from before and will automatically pull configurations depending on the name of the config. [`HCAConfig`](https://github.com/HumanCellAtlas/dcp-cli/blob/master/hca/config.py) forced the name to always be "hca" which meant that if ever you had run the command before (no matter when you called `HCAConfig()` things would already be initialized in an "hca"-y way and it would be unable to point to any other dss instance.

The easy solution---you say---just initialize with name not as "hca", [like this](https://github.com/DataBiosphere/cgp-dss-data-loader/commit/58add952296603b741b7585c63279b9eaa03af6d#diff-97306b10ea332a3669f64a09073cbdc3R80). It's a great idea, but the [`HCAConfig`](https://github.com/HumanCellAtlas/dcp-cli/blob/master/hca/config.py) code is written in such a way that doing this causes an error. the culpret is [line 9](https://github.com/HumanCellAtlas/dcp-cli/blob/master/hca/config.py#L9) where `name` is passed to `super()__init__` preventing us from overwriting it later down the inheritance tree.

To work around this I did a very hacky monkey patch to prevent setting this default `name="hca"`. This works great for our purposes; any previous "hca" configs are left untouched if the user had made some before, and we make fresh ones for ourselves.

Fixes #29, #35

┆Issue is synchronized with this [JIRA Task](https://ucsc-cgl.atlassian.net/browse/LOAD-42)
┆Project Name: CGP-Data Loader
┆Issue Number: LOAD-42

jessebrennan committed:
Work around HCA config bug (resolves STOR-2)

mikebaumann commented on tests/test_standard_loader_integration.py:
```diff
@@ -130,28 +130,25 @@ def _test_gen3_loading(self, test_json, bundle_guid, file_guid, file_version, mo
 
             message("Wait for newly loaded bundle to appear in search results")
-            search_results = self._search_for_bundle(bundle_guid)
+            bundle = self._get_bundle(bundle_guid)
```
Why did you make this change? The purpose of searching for the bundle is to ensure that that newly loaded bundle was successfully indexed, which is part of the loading process.
This changes you propose to this test are rejected.

jessebrennan replied:
I didn't realize why searching was important. Now that we are using auth, the bundles need to be accessed this way, but can be searched for as well to make sure they are indexed.

mikebaumann commented on util/__init__.py:
```diff
@@ -44,3 +45,5 @@ def utc_now():
     return datetime.datetime.utcnow().isoformat()
 
 
+def monkey_patch_hca_config():
```
This is an "interesting" way to resolve this problem.
What other options did you consider?

jessebrennan replied:
The ideal solution would be to put in a PR to HCA. This is an interim solution, but it works and it avoids messing with the internals of the config file. The problem was really old settings from HCA configs being used and then amended. If they were erased before, I imagine there would have been no config issues with the code. Doing the monkey patch allows us to just change the name of the config and avoid trying to manipulate the same set of configurations for so totally separate purposes.

mikebaumann replied:
Sounds good, I submitted this issue to HCA:
Problems accessing an alternate DSS from user scripts or unit tests #170
https://github.com/HumanCellAtlas/dcp-cli/issues/170

I have also changed calls to HCAConfig init to include parameters to not save the settings, to help prevent extraneous files from being created: `save_on_exit=True, autosave=False`

mikebaumann reviewed (changes requested):
Please address these comments.

Commit:
Changed test to use DSSClient to get bundle manifest

Changed test to use DSSClient to get bundle manifest instead of using
requests, so the authentication is performed automatically.

Commit:
Work around problems with initializing HCA DSSClient
